### PR TITLE
[Bug]: Type select / multiselect not updated in custom layouts

### LINKED ISSUE #14251: [Bug]: Type select / multiselect not updated in custom layouts
### Expected behavior

Update select options for field types select/multiselect when object class is modified

### Actual behavior

Select options are not updated, when options in a field in class is updated

### Steps to reproduce

The values in our class "Product" for field regions were wrong and were updated, but select option in the custom layouts are fixed to the old values. This breaks compatibility while editing in one custom layout and show / edit data in Master layout.

Select options in class after updating options:
image

Select options in custom layout after updating options:
image

Workarround: remove field from custom layout and add it again. As we have > 20 custom layouts, this is not the way to go, especially when the developer adds / removes options and don't care about custom layouts.

### PR DESCRIPTION
**Pull request by robertSt7**
resolves #14251

### CODE REVIEW

**Comment by lukmzig** [comment]
Hi @robertSt7, in general, this already looks quite good :) 
I just have a few points:
- In Advanced Many-To-Many Object Relation - I think we should not display Visible fields in the custom layouts (we are also not displaying them for Many-To-Many Object Relation. This can cause confusion as well, similliarly to the multiselect field options.
- When switching to the custom perspective and then back Master (Admin Mode) and exception is thrown.
`Message: Warning: Undefined array key "childs"
Trace: 
in /var/www/html/dev/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:591
#0 /var/www/html/dev/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php(588): Pimcore\Bundle\AdminBundle\Controller\Admin\DataObject\DataObjectController->injectValuesForCustomLayout(Array)
`
- In the type Select is currently not possible to define any options (looks same like in custom layouts)

Could you have a look please?

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 1]
```diff
@@ -567,12 +569,46 @@ static function (Task $task) {
...
+            } else {
+                foreach ($this->classFieldDefinitions[$child['name']] as $key => $value) {
+                    if ($child[$key] === null || $child[$key] === '' || (is_array($child[$key]) && empty($child[$key]))) {
+                        $child[$key] = $value;
+                    }
+                }
+            }
+        }
+    }
+
```
Comment (lukmzig) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:611]: Maybe you could combine this into one function and pass either children (for 11)  or childs

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectbricks.js [hunk 2]
```diff
@@ -51,25 +51,28 @@ pimcore.object.classes.data.objectbricks = Class.create(pimcore.object.classes.d
...
+        this.specificPanel.add({
+            xtype: "checkbox",
+            fieldLabel: t("border"),
+            name: "border",
+            checked: this.datax.border,
+        });
+
+
+        if (this.inCustomLayoutEditor) {
             this.specificPanel.add(new Ext.ux.form.MultiSelect({
```
Comment (lukmzig) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectbricks.js:76]: Is this field necessary at all in the custom layout editor?

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 3]
```diff
@@ -567,12 +569,46 @@ static function (Task $task) {
...
+                $this->injectValuesForCustomLayout($child);
+            } else {
+                foreach ($this->classFieldDefinitions[$child['name']] as $key => $value) {
+                    if ($child[$key] === null || $child[$key] === '' || (is_array($child[$key]) && empty($child[$key]))) {
+                        $child[$key] = $value;
+                    }
+                }
+            }
+        }
+    }
```
Comment (lukmzig) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php]: Maybe you could combine this into one function and pass either children (for 11)  or childs

File: bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectbricks.js (hunk 2 above)
Comment (robertSt7) [reply bundles/AdminBundle/Resources/public/js/pimcore/object/classes/data/objectbricks.js:76]: https://github.com/pimcore/pimcore/pull/12084 It is only implemented for custom layout, we may can think of another approach in an improvement.

**Review by lukmzig — APPROVED** [review]

**Pull request merged**